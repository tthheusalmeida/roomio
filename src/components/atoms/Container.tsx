import { mergeClassNames } from "@/utils/classNames";

export default function Container({
  children,
  className,
}: {
  children: React.ReactNode;
  className?: string;
}) {
  return (
    <div className={mergeClassNames(`container mx-auto px-4`, className)}>
      {children}
    </div>
  );
}
