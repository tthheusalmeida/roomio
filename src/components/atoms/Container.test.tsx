import { render, screen } from "@testing-library/react";
import Container from "./Container";

describe("Container component", () => {
  it("renders children correctly", () => {
    const { asFragment } = render(
      <Container>
        <span>Content inside</span>
      </Container>
    );

    expect(screen.getByText("Content inside")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <Container className="extra-class">
        <span>Content</span>
      </Container>
    );

    const div = screen.getByText("Content").parentElement;
    expect(div).toHaveClass("w-full", "mx-auto", "px-4");
    expect(div).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
