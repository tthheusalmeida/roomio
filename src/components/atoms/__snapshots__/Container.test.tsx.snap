// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Container component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <div
    class="w-full mx-auto px-4 extra-class"
  >
    <span>
      Content
    </span>
  </div>
</DocumentFragment>
`;

exports[`Container component renders children correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full mx-auto px-4"
  >
    <span>
      Content inside
    </span>
  </div>
</DocumentFragment>
`;
